### PR TITLE
new asset file formatting

### PR DESCRIPTION
**Pull request by craigmulligan**
Fixes formatting changes from the `1.1.2` release.

### CODE REVIEW

**Comment by lurch** [comment]
@craig-mulligan Looks like `getArch` and `getArchString` are behaving slightly differently - https://etcher.io is now saying `Etcher for Linux x86 (32-bit)` (correct) but `Etcher for Linux x86 (64-bit)` (incorrect).
Perhaps the two methods `getArch` and `getArchString` should be modified so that one of them calls the other, so that the logic doesn't have to be duplicated in two places ;-)

**Comment by craigmulligan** [comment]
@lurch yea this script has be modified as needed so could definitely be cleaner. It'd be super useful if you could give me a list of the releases that should be listed on the site and the plain text title for each.

**Comment by lurch** [comment]
Okay, this is (IMHO - maybe @jviotti would like to comment too?) what it should display :

| Filename | DisplayText | OS | Arch |
| --- | --- | --- | --- |
| `Etcher-Setup-1.1.2-x64.exe` | Etcher for Windows x64 (64-bit) (Installer) | Windows | x64 |
| `Etcher-Portable-1.1.2-x64.exe` | Etcher for Windows x64 (64-bit) (Portable) | Windows | x64 |
| `Etcher-Setup-1.1.2-x86.exe` | Etcher for Windows x86 (32-bit) (Installer) | Windows | x86 |
| `Etcher-Portable-1.1.2-x86.exe` | Etcher for Windows x86 (32-bit) (Portable) | Windows | x86 |
| `Etcher-1.1.2.dmg` | Etcher for OS X x64 | OS X | x64 |
| `etcher-1.1.2-linux-x86_64.zip` | Etcher for Linux x64 (64-bit) | Linux | x64 |
| `etcher-1.1.2-linux-i386.zip` | Etcher for Linux x86 (32-bit) | Linux | x86 | 

...and ideally in this order too i.e. Always display Windows first, then OSX, then Linux. Always display 64-bit first, then 32-bit. Always display Installer-version first (if applicable), then Portable-version.

(given that OS X doesn't support anything _other_ than x64 (AFAIK), I wonder if the "Etcher for OS X x64" label should be just "Etcher for OS X"?)

@jviotti Do you think it'd be worth suffixing the Linux zipfile DisplayTexts with ` (AppImage)` ?

@craig-mulligan These filename-changes were due to https://github.com/resin-io/etcher/pull/1609  Hopefully with a bit of luck the filename formats won't be changing again next time ;-)

**Comment by jviotti** [comment]
@lurch's suggestions sound great. I agree with the ordering, the removal of x64 for OS X and the AppImage indication.

Just one extra comment: can we say macOS instead of OS X?

**Comment by lurch** [comment]
> Just one extra comment: can we say macOS instead of OS X?

https://en.wikipedia.org/wiki/MacOS says "**macOS** (previously **Mac OS X** and later **OS X**)" - interesting, I wasn't aware of that ;-)

**Comment by craigmulligan** [comment]
@lurch @jviotti, 

Thanks a stack for the table super helpful 👍 , I've update the PR according to the table and deployed the site so you can see it live. Let me know if there is anything to changes to be made before merging.

File: lib/getLatestRelease.js [hunk 1]
```diff
@@ -13,58 +14,79 @@ if (process.env.GH_TOKEN) {
...
-  if (!includes(file, '.exe')) {
-    return '';
-  }
-
-  if (includes(file, 'Portable')) {
-    return '(Portable)';
-  } else {
-    return '(Installer)';
+const getArch = str => {
+  if (includes(str, '64') || getOS(str) === 'macOS') {
```
Comment (lurch) [inline lib/getLatestRelease.js]: IMHO this is a bit too generic - and it'll go horribly wrong if we get up to Etcher version 1.1.64 ? ;-)
What about:
```js
if (includes(str, 'x64') || includes(str, 'x86_64') || includes(str, 'amd64') || getOS(str) === 'macOS') {
```
(included `amd64` as forward-planning, see https://github.com/resin-io/etcher/releases )

File: lib/getLatestRelease.js [hunk 2]
```diff
@@ -13,58 +14,79 @@ if (process.env.GH_TOKEN) {
 
 const getOS = str => {
   if (includes(str, 'darwin') || includes(str, '.dmg')) {
```
Comment (lurch) [inline lib/getLatestRelease.js]: Could we also add a `includes(str, 'mac') || ` here? (see https://github.com/resin-io/etcher/releases )

EDIT: And then we can also re-use this function down on line 98 :smiley:

File: lib/getLatestRelease.js [hunk 3]
```diff
@@ -13,58 +14,79 @@ if (process.env.GH_TOKEN) {
...
-  return '';
+  return 'x86';
 };
 
 const getArchString = str => {
-  if (includes(str, 'x86_64') || includes(str, 'x64')) {
-    return '(64-bit)';
-  } else if (includes(str, 'i386') || includes(str, 'x86')) {
-    return '(32-bit)';
+  switch (getArch(str)) {
```
Comment (lurch) [inline lib/getLatestRelease.js]: :+1:

File: lib/getLatestRelease.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import includes from 'lodash/includes';
+import _ from 'lodash';
```
Comment (lurch) [inline lib/getLatestRelease.js]: Looks like you don't actually use this?

File: lib/getLatestRelease.js [hunk 5]
```diff
@@ -75,13 +95,14 @@ const prepAssets = assets => {
...
         includes(item.name, 'electron')
       );
     })
     .map(item => {
       return {
-        text: prettifyFileName(item.name),
+        text: packagePrettyName(item.name),
         href: item.browser_download_url,
         os: getOS(item.name),
         arch: getArch(item.name)
```
Comment (lurch) [inline lib/getLatestRelease.js]: Just to be on the safe side, it might be worth lower-casing `item.name` everywhere where it's used?

**Review by lurch — COMMENTED** [review]
Looking great @craig-mulligan :+1: 
Could you tweak it so that Installer gets listed before Portable though?
(also added a few comments)

**Comment by craigmulligan** [comment]
@lurch cheers for the help, I've adjusted accordingly, mind taking another look?

**Review by lurch — APPROVED** [review]
Looks perfect @craig-mulligan :1st_place_medal: 
My only remaining (optional) suggestion would be to move the `AppImage` stuff inside `getInstallerType`

**Comment by lurch** [comment]
Thanks Craig :-)

**Pull request merged**